pkg/service/model_build_target_group.go

```diff
@@ -215,12 +215,18 @@ func (t *defaultModelBuildTask) buildTargetGroupAttributes(_ context.Context) ([
 	if _, ok := rawAttributes[tgAttrsProxyProtocolV2Enabled]; !ok {
 		rawAttributes[tgAttrsProxyProtocolV2Enabled] = strconv.FormatBool(t.defaultProxyProtocolV2Enabled)
 	}
-	proxyV2Annotation := ""
-	if exists := t.annotationParser.ParseStringAnnotation(annotations.SvcLBSuffixProxyProtocol, &proxyV2Annotation, t.service.Annotations); exists {
-		if proxyV2Annotation != "*" {
-			return []elbv2model.TargetGroupAttribute{}, errors.Errorf("invalid value %v for Load Balancer proxy protocol v2 annotation, only value currently supported is *", proxyV2Annotation)
+	var proxyV2Annotations []string
+	if exists := t.annotationParser.ParseStringSliceAnnotation(annotations.SvcLBSuffixProxyProtocol, &proxyV2Annotations, t.service.Annotations); exists {
+		for _, proxySelector := range proxyV2Annotations {
+			if proxySelector == "*" {
+				rawAttributes[tgAttrsProxyProtocolV2Enabled] = "true"
+				break
+			}
+			if proxySelector == strconv.Itoa(int(port.Port)) {
```

Comment:
iiuc, if one port in the TG is in the annotation to enable proxy protocol, this TG will have proxy protocol v2 enabled? Maybe need to call out in the doc too.
Also, I'm just thinking about some edge cases, do we need to add a nil check for `port.Port`?

Reply:
@arnaud-dezandee

Reply:
I'll check that yes

Reply:
just checked, nil check is not required here

Reply:
I added more details in the doc